### PR TITLE
Report data hub

### LINKED ISSUE #434: ReportHub
-Objective: Compute ReportVariable and return IDataCube<ReportVariable>
-Required Data Hubs: ReferenceDataHub(only some type (ReportingNode, others?)), ParameterHub(only FxParameter?),IfrsVariableHub)
-Computation: it requires ReportStorage (+called methods) and ReportScopes

Test:
Compute ReportVariable DataCube for a simple reportType (PV). 
Assert that the properties of the report variables are correctly populated (check the join, a prop from IfrsVar and one from GoC)
Compute ReportVariable DataCube for a complex ReportType (PnL)
Assert existence of variables.

### PR DESCRIPTION
**Pull request by amuolo**

### CODE REVIEW

File: OpenSmc.Ifrs17/ReportHub/ReportHubConfiguration.cs [hunk 1]
```diff
@@ -1,52 +1,95 @@
...
+                    .FromHub(parameterAddress, ds => ds 
+                        .WithType<ExchangeRate>().WithType<CreditDefaultRate>().WithType<PartnerRating>())
+                    .FromHub(ifrsVarAddress, ds => ds
+                        .WithType<IfrsVariable>())
+                    .AddCustomInitialization(ReportInit(config))))
+
+            .WithRoutes(route => route.RouteMessage<GetManyRequest<ReportVariable>>(_ => reportAddress));
+    }
+
+    public static Action<HubDataSource, ScopeFactory> ReportInit(MessageHubConfiguration config)
```
Comment (dtrzesniak) [inline OpenSmc.Ifrs17/ReportHub/ReportHubConfiguration.cs:62]: What is HubDataSource?

**Review by dtrzesniak — APPROVED** [review]

**Comment by dtrzesniak** [comment]
Maybe complete #440 first...

File: OpenSmc.Ifrs17/ReportHub/ReportHubConfiguration.cs (hunk 1 above)
Comment (amuolo) [reply OpenSmc.Ifrs17/ReportHub/ReportHubConfiguration.cs:62]: The data plugin has a HubDataSource inside to manage the data. And here I extract this in order to calculate ReportVariables.

**Comment by amuolo** [comment]
> Maybe complete #440 first...

yes, also in my opinion #440 can be completed first. @kalabind could you please complete it?

**Comment by kalabind** [comment]
> > Maybe complete #440 first...
> 
> yes, also in my opinion #440 can be completed first. @kalabind could you please complete it?

Yep. Completed #440. Sorry for delay @amuolo

**Pull request merged**